app/controllers/application_controller/authentication_dependency.rb

```diff
@@ -5,7 +5,7 @@ class ApplicationController
 
   def current_scopes
     return [] unless logged_in?
-    session[:current_scopes] ||= current_user.github_client_scopes
+    session[:current_scopes] = current_user.github_client_scopes
```

Comment:
Changed to just use `=` so that we don't just keep adding to scopes

Reply:
> so that we don't just keep adding to scopes

What do you mean by this?

`||=` is usually used for simple caching, `session[:current_scopes] ||= current_user.github_client_scopes` returns `session[:current_scopes]` if it's not nil, and otherwise, it sets `session[:current_scopes]` and returns `current_user.github_client_scopes`.

Reply:
With the `||=` operator the current_scopes did not return the right value if the user ever reduced their scopes (for example if the token was invalidated and valid scopes went to empty). In that case we just kept using the cached value of the scopes since it wasn't actually nil.

Reply:
Ok, sounds good :+1: